cfme/intelligence/reports/menus.py

```diff
@@ -180,6 +180,30 @@ def manage_folder(self, group, folder=None):
             view.manager.commit()
             view.save_button.click()
 
+    def move_reports(self, group, folder, subfolder, reports):
+        """ Moves a list of reports to a given menu
+        Args:
+            group: User group
+            folder: Parent of the subfolder where reports are to be moved.
+            subfolder: Subfolder under which the reports are to be moved.
+            reports: List of reports that are to be moved.
+        """
+        if not isinstance(reports, list):
+            reports = [reports]
+
+        with self.manage_subfolder(group, folder, subfolder) as selected_menu:
+            selected_options = selected_menu.parent_view.report_select.all_options
+            for report in reports:
+                if report in selected_options:
+                    raise Exception("Report already present. Cannot move.")
```

Comment:
Thoughts on making a custom expression for this scenario, e.g. `ReportCannotBeMovedException`, also some information about where the report is already present might be helpful.

Reply:
Definitely! That's a good idea! I'll see if I can implement that.

Reply:
Actually, this error is raised when the report is already present in the place where it is being moved. I'll modify the exception message to make things clearer.